docs/pipelines/architecture/index.md

```diff
@@ -0,0 +1,29 @@
+# Architecture
+
+There are three components in pipelines - the orchestrator, the dispatcher, and the executor. The orchestrator determines the jobs that needs to be executed, the dispatcher takes each job as input and calls the executor then reports the status of the job. Each component is explained in detail in sections below.
+
+## Orchestrator
+
+The orchestrator identifies each infrastructure change in the Pull Request or git commit, classifies the type of infrastructure change it is (e.g. AccountAdded, ModuleChanged, EnvCommonChanged) and determines the right pipelines actions (e.g., terragrunt plan, apply, or destroy) to run based on that infrastructure change. For each infrastructure change, it calls the dispatcher.
```

Comment:
```suggestion
The orchestrator identifies each infrastructure change in the pull request or git commit, classifies the type of infrastructure change it is (e.g. AccountAdded, ModuleChanged, EnvCommonChanged), and determines the right pipelines actions (e.g., `terragrunt plan`, `apply`, or `destroy`) to run based on that infrastructure change. For each infrastructure change, it calls the dispatcher.
```

Are we proposing "AccountAdded" etc. as formal terms? I'm wondering about the CamelCase here, or whether we should just use "e.g. account added, module changed, module defaults changed, etc." In either case, we shouldn't use "EnvCommonChanged" anymore.

Reply:
I like them as formal terms, yes. It introduces the infra change type as a first class concept. I can change the env common bit.